dynamiq/nodes/tools/http_api_call.py

```diff
@@ -43,6 +43,58 @@ def validate_dict_fields(cls, value: Any, field: str) -> Any:
             raise ActionParsingException(f"Expected a dictionary or a JSON string for '{field}'.")
 
 
+def recursive_merge(dict1: dict, dict2: dict) -> dict:
+    """
+    Recursively merge two dictionaries.
+
+    If a key exists in both dictionaries:
+    - If value are not list or dict, the value in dict1 will be used.
+    - If both values are list or dict, they are concatenated.
+
+    Args:
+        dict1 (dict): The first dictionary to merge.
+        dict2 (dict): The second dictionary to merge.
+
+    Returns:
+        dict: The merged dictionary
+    """
+    for key, value in dict2.items():
+        if key in dict1:
+            if isinstance(dict1[key], dict) and isinstance(value, dict):
+                dict1[key] = recursive_merge(dict1[key], value)
+            elif isinstance(dict1[key], list) and isinstance(value, list):
+                dict1[key].extend(x for x in value if x not in set(dict1[key]))
+        else:
+            dict1[key] = value
+    return dict1
+
+
+def concatenate_parameters(
+    input_params: HttpApiCallInputSchema, additional_params: HttpApiCallInputSchema
+) -> HttpApiCallInputSchema:
+    """
+    Concatenate parameters for an API call.
+
+    Args:
+        input_params (HttpApiCallInputSchema): The first set of parameters for the API call.
+        additional_params (HttpApiCallInputSchema): The second set of parameters to be merged into the first.
+
+    Returns:
+        HttpApiCallInputSchema: A new object with the combined parameters.
+    """
+    combined_params = recursive_merge(input_params.params.copy(), additional_params.params)
+    combined_headers = recursive_merge(input_params.headers.copy(), additional_params.headers)
+    combined_data = recursive_merge(input_params.data.copy(), additional_params.data)
```

Comment:
```suggestion
    combined_params = recursive_merge(input_params.params.copy(), additional_params.params)
    combined_headers = recursive_merge(input_params.headers.copy(), additional_params.headers)
    combined_data = recursive_merge(input_params.data.copy(), additional_params.data)
```

Consider moving the copy logic inside the function to avoid having duplicated code for copying